src/com/jambit/STDatabase.java

```diff
@@ -0,0 +1,22 @@
+package com.jambit;
+
+import java.io.IOException;
+import java.sql.SQLException;
+
+public class STDatabase {
+  private static Database database = null;
+
+  private STDatabase() {}
+
+  public static Database getInstance()
+      throws Database.MultipleConnectionException, IOException, SQLException,
+          Database.TimeOutException {
+    if (database == null) {
+      database = new Database();
+      database.connect();
+    } else {
+      database.checkValid();
```

Comment:
this check is useless. You check something but you ignore the return value. Hint: Why do you make this check?